# eheheh sir
### cleaningings datings 


In [1]:
import pandas as pd

In [2]:
df1 = pd.read_csv('C:/Users/extra/Downloads/Data/Google Data Analytics (Cyclistic) Capstone Project/divvy-tripdata.csv', index_col=0)
df1

C:\Users\extra\AppData\Local\Temp\ipykernel_8248\1212645200.py:1: DtypeWarning: Columns (5,7) have mixed types. Specify dtype option on import or set low_memory=False.
  df1 = pd.read_csv('C:/Users/extra/Downloads/Data/Google Data Analytics (Cyclistic) Capstone Project/divvy-tripdata.csv', index_col=0)


rideable_type           started_at             ended_at  \
ride_id                                                                     
A847FADBBC638E45    docked_bike  2020-04-26 17:45:14  2020-04-26 18:12:03   
5405B80E996FF60D    docked_bike  2020-04-17 17:08:54  2020-04-17 17:17:03   
5DD24A79A4E006F4    docked_bike  2020-04-01 17:54:13  2020-04-01 18:08:36   
2A59BBDF5CDBA725    docked_bike  2020-04-07 12:50:19  2020-04-07 13:02:31   
27AD306C119C6158    docked_bike  2020-04-18 10:22:59  2020-04-18 11:15:54   
...                         ...                  ...                  ...   
C349E243A9BAA6F7  electric_bike  2022-11-25 11:19:52  2022-11-25 11:31:50   
B0B4E85DA43A9194   classic_bike  2022-11-22 16:57:53  2022-11-22 17:31:29   
8D148DD47B59530B   classic_bike  2022-11-06 13:04:05  2022-11-06 13:13:33   
0D1170BA18FD33D1   classic_bike  2022-11-06 09:41:29  2022-11-06 15:17:17   
09B20DC75B5EA1E0  electric_bike  2022-11-26 11:59:28  2022-11-26 12:31:04   

                             start_station_name start_station_id  \
ride_id                                                            
A847FADBBC638E45                   Eckhart Park               86   
5405B80E996FF60D      Drake Ave & Fullerton Ave              503   
5DD24A79A4E006F4           McClurg Ct & Erie St              142   
2A59BBDF5CDBA725   California Ave & Division St              216   
27AD306C119C6158           Rush St & Hubbard St              125   
...                                         ...              ...   
C349E243A9BAA6F7         Wabash Ave & Grand Ave     TA1307000117   
B0B4E85DA43A9194     Franklin St & Jackson Blvd     TA1305000025   
8D148DD47B59530B  Michigan Ave & Ida B Wells Dr     TA1305000010   
0D1170BA18FD33D1           Halsted St & 18th St            13099   
09B20DC75B5EA1E0  Michigan Ave & Ida B Wells Dr     TA1305000010   

                                end_station_name end_station_id  start_lat  \
ride_id                                                                      
A847FADBBC638E45     Lincoln Ave & Diversey Pkwy          152.0  41.896400   
5405B80E996FF60D                 Kosciuszko Park          499.0  41.924400   
5DD24A79A4E006F4      Indiana Ave & Roosevelt Rd          255.0  41.894500   
2A59BBDF5CDBA725          Wood St & Augusta Blvd          657.0  41.903000   
27AD306C119C6158      Sheridan Rd & Lawrence Ave          323.0  41.890200   
...                                          ...            ...        ...   
C349E243A9BAA6F7                             NaN            NaN  41.891836   
B0B4E85DA43A9194  Sheffield Ave & Wrightwood Ave   TA1309000023  41.877708   
8D148DD47B59530B                  Shedd Aquarium          15544  41.876243   
0D1170BA18FD33D1                  Shedd Aquarium          15544  41.857506   
09B20DC75B5EA1E0                  Shedd Aquarium          15544  41.876272   

                  start_lng    end_lat    end_lng member_casual  
ride_id                                                          
A847FADBBC638E45 -87.661000  41.932200 -87.658600        member  
5405B80E996FF60D -87.715400  41.930600 -87.723800        member  
5DD24A79A4E006F4 -87.617900  41.867900 -87.623000        member  
2A59BBDF5CDBA725 -87.697500  41.899200 -87.672200        member  
27AD306C119C6158 -87.626200  41.969500 -87.654700        casual  
...                     ...        ...        ...           ...  
C349E243A9BAA6F7 -87.626806  41.890000 -87.620000        member  
B0B4E85DA43A9194 -87.635321  41.928712 -87.653833        casual  
8D148DD47B59530B -87.624426  41.867226 -87.615355        casual  
0D1170BA18FD33D1 -87.645991  41.867226 -87.615355        casual  
09B20DC75B5EA1E0 -87.624576  41.867226 -87.615355        casual  

[14195770 rows x 12 columns]

In [3]:
#removing empty values
df12 = df1.dropna()

In [4]:
#checking if all empty values were removed
df12.isna()

rideable_type  started_at  ended_at  start_station_name  \
ride_id                                                                     
A847FADBBC638E45          False       False     False               False   
5405B80E996FF60D          False       False     False               False   
5DD24A79A4E006F4          False       False     False               False   
2A59BBDF5CDBA725          False       False     False               False   
27AD306C119C6158          False       False     False               False   
...                         ...         ...       ...                 ...   
03005DDB0F2F28E4          False       False     False               False   
B0B4E85DA43A9194          False       False     False               False   
8D148DD47B59530B          False       False     False               False   
0D1170BA18FD33D1          False       False     False               False   
09B20DC75B5EA1E0          False       False     False               False   

                  start_station_id  end_station_name  end_station_id  \
ride_id                                                                
A847FADBBC638E45             False             False           False   
5405B80E996FF60D             False             False           False   
5DD24A79A4E006F4             False             False           False   
2A59BBDF5CDBA725             False             False           False   
27AD306C119C6158             False             False           False   
...                            ...               ...             ...   
03005DDB0F2F28E4             False             False           False   
B0B4E85DA43A9194             False             False           False   
8D148DD47B59530B             False             False           False   
0D1170BA18FD33D1             False             False           False   
09B20DC75B5EA1E0             False             False           False   

                  start_lat  start_lng  end_lat  end_lng  member_casual  
ride_id                                                                  
A847FADBBC638E45      False      False    False    False          False  
5405B80E996FF60D      False      False    False    False          False  
5DD24A79A4E006F4      False      False    False    False          False  
2A59BBDF5CDBA725      False      False    False    False          False  
27AD306C119C6158      False      False    False    False          False  
...                     ...        ...      ...      ...            ...  
03005DDB0F2F28E4      False      False    False    False          False  
B0B4E85DA43A9194      False      False    False    False          False  
8D148DD47B59530B      False      False    False    False          False  
0D1170BA18FD33D1      False      False    False    False          False  
09B20DC75B5EA1E0      False      False    False    False          False  

[11784754 rows x 12 columns]

In [5]:
df12.dtypes

rideable_type          object
started_at             object
ended_at               object
start_station_name     object
start_station_id       object
end_station_name       object
end_station_id         object
start_lat             float64
start_lng             float64
end_lat               float64
end_lng               float64
member_casual          object
dtype: object

In [6]:
#listing types of rides
df12.rideable_type.unique()

array(['docked_bike', 'electric_bike', 'classic_bike'], dtype=object)

In [7]:
df12.rideable_type.value_counts()

classic_bike     5836546
docked_bike      3020292
electric_bike    2927916
Name: rideable_type, dtype: int64

### Types and number of rides
1. classic_bike     5836546
2. docked_bike      3020292
3. electric_bike    2927916

In [8]:
df12.member_casual.unique()

array(['member', 'casual'], dtype=object)

In [9]:
df12.member_casual.value_counts()

member    6757397
casual    5027357
Name: member_casual, dtype: int64

### Types and number of customers

1. member    6757397
2. casual    5027357